WordPress/src/main/java/org/wordpress/android/ui/mysite/SelectedSiteSource.kt

```diff
@@ -17,26 +18,24 @@ import javax.inject.Singleton
 class SelectedSiteSource @Inject constructor(
     private val selectedSiteRepository: SelectedSiteRepository,
     private val dispatcher: Dispatcher
-) : MySiteSource<SelectedSite> {
-    val refresh: MutableLiveData<Boolean> = MutableLiveData<Boolean>(false)
+) : MySiteRefreshSource<SelectedSite> {
+    override val refresh: MutableLiveData<Boolean> = MutableLiveData<Boolean>(false)
 
-    override fun buildSource(
+    override fun build(
         coroutineScope: CoroutineScope,
         siteLocalId: Int
     ) = selectedSiteRepository.selectedSiteChange
             .filter { it == null || it.id == siteLocalId }
             .map { SelectedSite(it) }
 
-    fun refresh() {
+    override fun refresh() {
         selectedSiteRepository.updateSiteSettingsIfNecessary()
         selectedSiteRepository.getSelectedSite()?.let {
             refresh.postValue(true)
```

Comment:
Suggestion (💡): How about using the `super.refresh()` here to make sure that if in the future the generic `refresh()` interface implementation changes then we would be safe as with that this will be updated automatically as well?

Reply:
hmmm .. do you think `super.refresh()` would look out-of-place within the `?.let` -- that was my only hesitation. 
@ParaskP7

Reply:
I added it, I can live with the awkwardness if you can. 👍

Reply:
> I can live with the awkwardness if you can.

🤣 + 🙇